File: HangJungCho/2주차/PG_ 42576_완주하지못한선수.cpp
/*
기본접근
중복이 될 수 있으므로 unordered_multiset을 활용하여 조회한다.
선수 목록을 1개씩 받아와서 완주목록에서 조회후 있으면 제거, 없을경우 return
*/

#include <string>
#include <vector>
#include <set>
#include <unordered_set>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_multiset<string> Myset(completion.begin(), completion.end());
    for(int i=0;i<participant.size();i++){
        auto it = Myset.find(participant[i]);
        if (it != Myset.end()) { Myset.erase(it);}
        else
            return participant[i];
    }
}

File: Seoin0319/2_week/PG_42576_완주하지못한선수.py
def solution(participant, completion):
    # 1. 참가자와 완주자 이름을 각각 정렬
    participant.sort()
    completion.sort()
    
    # 2. 완주자 명단과 순서대로 비교
    for i in range(len(completion)):
        # 이름이 다르면 현재 참가자가 완주하지 못한 사람
        if participant[i] != completion[i]:
            return participant[i]
    
    # 3. 여기까지 전부 같으면,
    #    참가자 명단의 마지막 사람이 완주하지 못한 사람
    return participant[-1]